src/helics/shared_api_library/helicsCore.h

```diff
@@ -1007,24 +1007,58 @@ HELICS_EXPORT void helicsFederateEnterInitializingMode(HelicsFederate fed, Helic
 HELICS_EXPORT void helicsFederateEnterInitializingModeAsync(HelicsFederate fed, HelicsError* err);
 
 /**
- * Check if the current Asynchronous operation has completed.
+ * Complete the entry to initialize mode that was initiated with /ref heliceEnterInitializingModeAsync.
+ *
+ * @param fed The federate desiring to complete the initialization step.
+ *
+ * @param[in,out] err An error object that will contain an error code and string if any error occurred during the execution of the function.
+ */
+HELICS_EXPORT void helicsFederateEnterInitializingModeComplete(HelicsFederate fed, HelicsError* err);
+
+/**
+ * Trigger a blocking call and return to created state after all federates have either triggered an iteration or waiting to enter
+ * initializing mode
+ *
+ * @details this call will return the federate to the created state to allow additional setup to occur with federates either iterating in
+ * the mode or waiting.
  *
  * @param fed The federate to operate on.
  *
  * @param[in,out] err An error object that will contain an error code and string if any error occurred during the execution of the function.
+ */
+HELICS_EXPORT void helicsFederateEnterInitializingModeIterative(HelicsFederate fed, HelicsError* err);
+
+/**
+ * Non blocking alternative to \ref helicsFederateEnterInitializingModeIterative.
  *
- * @return HELICS_FALSE if not completed, HELICS_TRUE if completed.
+ * @details The function helicsFederateEnterInitializationModeIterativeComplete must be called to finish the operation.
+ *
+ * @param fed The federate to operate on.
+ *
+ * @param[in,out] err An error object that will contain an error code and string if any error occurred during the execution of the function.
  */
-HELICS_EXPORT HelicsBool helicsFederateIsAsyncOperationCompleted(HelicsFederate fed, HelicsError* err);
+HELICS_EXPORT void helicsFederateEnterInitializingModeIterativeAsync(HelicsFederate fed, HelicsError* err);
 
 /**
- * Finalize the entry to initialize mode that was initiated with /ref heliceEnterInitializingModeAsync.
+ * Complete the call to initialize mode that was initiated with /ref heliceEnterInitializingModeIterativeAsync.  The federate will be in
+ * startup or error mode on return
  *
- * @param fed The federate desiring to complete the initialization step.
+ * @param fed The federate desiring to iterate to the startup mode again
```

Comment:
```suggestion
 * @param fed The federate that was used to call /ref helicsEnterInitializingModeIterativeAsync
```

The way I'm reading the current comment, I can provide any federate as the parameter even if it didn't call the async function, and this signals that the federate wishes to go through another iteration before entering initialization mode.

I think the use of "startup" mode is confusing -- it seems to be the same thing as initializing mode?

Reply:
I will see if I can rework the doc to be clearer